consensus/cryptarchia-engine/src/lib.rs

```diff
@@ -1,157 +1,163 @@
-pub mod block;
 pub mod config;
-pub mod crypto;
-pub mod leader_proof;
-pub mod ledger;
-pub mod time;
 
-pub use block::*;
+mod time;
+
 pub use config::*;
-pub use leader_proof::*;
-use ledger::{Ledger, LedgerState};
 use std::collections::{HashMap, HashSet};
 use thiserror::Error;
-pub use time::*;
+pub use time::{Epoch, Slot};
 
 #[derive(Clone, Debug)]
-pub struct Cryptarchia {
-    local_chain: Branch,
-    branches: Branches,
-    ledger: Ledger,
```

Comment:
Prior to this PR, we had this `Cryptarchia` struct tying together ledger and branches.

It was making sure that whenever branches was updated, the ledger was also updated (and vice-versa).

This functionality seems to have been dropped in this PR? Or I'm just now seeing where it moved.

It seems we need something like this `Cryptarchia` struct in `ledger` to keep branches and ledger in sync.